### PR TITLE
Explicitly define `Pipeline` and `ISBSvc` in the Rollout CRDs

### LINKED ISSUE #119: Explicitly define Pipeline and ISBSvc in `PipelineRollout` and `ISBServiceRollout`  CRD
# Summary

Currently, PipelineRollout CR embed Pipeline Spec directly as `runtime.RawExtension` with the naming of [Pipeline](https://github.com/numaproj/numaplane/blob/main/pkg/apis/numaplane/v1alpha1/pipelinerollout_types.go#L31). It can confuse user this is only includes the Pipeline Spec but not the whole Pipeline definition. Similarly for ISBService. We should update the CRD to make it straightforward. 

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

### PR DESCRIPTION
**Pull request by xdevxy**
Fixes #119 

### Modifications

This PR explicitly define `Pipeline` and `ISBSvc` in the Rollout CRDs to make it clear what are the fields are used in Rollout CRDs. Currently only add `Spec` field.

### Verification
`make test`

### CODE REVIEW

File: pkg/apis/numaplane/v1alpha1/isbservicerollout_types.go [hunk 1]
```diff
@@ -26,7 +26,12 @@ import (
 
 // ISBServiceRolloutSpec defines the desired state of ISBServiceRollout
 type ISBServiceRolloutSpec struct {
-	InterStepBufferService runtime.RawExtension `json:"interStepBufferService"`
+	InterStepBufferService InterStepBufferService `json:"InterStepBufferService"`
```
Comment (juliev0) [inline pkg/apis/numaplane/v1alpha1/isbservicerollout_types.go:29]:
```suggestion
	InterStepBufferService InterStepBufferService `json:"interStepBufferService"`
```

**Comment by juliev0** [comment]
I was going to ask if you verified that pausing still works, but I guess our unit test effectively does that since it verifies that the pipeline [got updated](https://github.com/numaproj/numaplane/blob/main/internal/controller/pipelinerollout_controller_test.go#L178) (which would only happen if the pausing worked).

**Review by KeranYang — COMMENTED** [review]
Thanks, Hao for making the quick change. Can we also update the [samples](https://github.com/numaproj/numaplane/tree/main/config/samples) to reflect the change?

File: pkg/apis/numaplane/v1alpha1/isbservicerollout_types.go (hunk 1 above)
Comment (juliev0) [reply pkg/apis/numaplane/v1alpha1/isbservicerollout_types.go:29]: shouldn't this be lower case?

Comment (KeranYang) [reply pkg/apis/numaplane/v1alpha1/isbservicerollout_types.go:29]: Yes, it should match the numaflow version of it.

Comment (KeranYang) [reply pkg/apis/numaplane/v1alpha1/isbservicerollout_types.go]: https://github.com/numaproj/numaflow/blob/main/pkg/apis/numaflow/v1alpha1/pipeline_types.go#L453

Comment (KeranYang) [reply pkg/apis/numaplane/v1alpha1/isbservicerollout_types.go]: taking back my response, this one doesn't need to match. The service name one needs to.

**Review by afugazzotto — APPROVED** [review]
LGTM

**Review by juliev0 — APPROVED** [review]

**Review by KeranYang — APPROVED** [review]

**Pull request merged**